charms/worker/k8s/src/token_distributor.py

```diff
@@ -23,6 +23,24 @@
 SECRET_ID = "{0}-secret-id"  # nosec
 
 
+class K8sCharm(Protocol):
+    """Typing for the K8sCharm.
+
+    Attributes:
+        app (ops.Application): The application object.
+        model (ops.Model): The model object.
+        unit (ops.Unit): The unit object.
+    """
+
+    app: ops.Application
+    model: ops.Model
+    unit: ops.Unit
+
+    def get_cluster_name(self) -> str:
+        """Get the cluster name."""
+        ...  # pylint: disable=unnecessary-ellipsis
+
+
```

Comment:
I can't `import charm` in here b/c that'd create a circular dependency.  So lets make a protocol for the things in this module we use from the `K8sCharm` type